arangod/GeneralServer/RestHandler.cpp

```diff
@@ -230,23 +230,32 @@ futures::Future<Result> RestHandler::forwardRequest(bool& forwarded) {
   }
 
   options.acceptType = rest::contentTypeToString(_request->contentTypeResponse());
-    
+
   for (auto const& i : _request->values()) {
     options.param(i.first, i.second);
   }
-  
+
   auto requestType =
       fuerte::from_string(GeneralRequest::translateMethod(_request->requestType()));
 
   VPackStringRef resPayload = _request->rawPayload();
   VPackBuffer<uint8_t> payload(resPayload.size());
   payload.append(resPayload.data(), resPayload.size());
-  
+
   nf.trackForwardedRequest();
- 
-  auto future = network::sendRequest(pool, "server:" + serverId, requestType,
-                                     _request->requestPath(),
-                                     std::move(payload), options, std::move(headers));
+
+  // Should the coordinator be gone by now, we'll respond with 404.
+  // There is no point forwarding requests. This affects transactions, cursors, ...
+  if (server().getFeature<ClusterFeature>().clusterInfo().getServerEndpoint(serverId).empty()) {
+    generateError(rest::ResponseCode::NOT_FOUND,
+                  TRI_ERROR_CLUSTER_SERVER_UNKNOWN,
+                  std::string("cluster server ") + serverId + " unknown");
+    return Result(TRI_ERROR_CLUSTER_SERVER_UNKNOWN);
+  }
```

Comment:
Yes, this can happen, but it does not matter. In such a case we would hand out 503 as before. That is OK, since it will probably let a client do a retry and we are back here. It is only necessary that we will **eventually** return 404, such that the client (in our case the testagent) does not retry indefinitely.